lib/src/Base/Optim/Cobyla.cxx

```diff
@@ -123,7 +123,7 @@ void Cobyla::run()
    */
   int returnCode = ot_cobyla(n, m, &x[0], rhoBeg_, rhoEnd, message, &maxFun, Cobyla::ComputeObjectiveAndConstraint, (void*) this);
 
-  result_ = OptimizationResult(dimension, 2);
+  result_ = OptimizationResult(dimension, 1);
```

Comment:
Looks wrong to me, see l.101
```
  // initialize history
  evaluationInputHistory_ = Sample(0, dimension);
  evaluationOutputHistory_ = Sample(0, 2);
```

Reply:
Okay, got it now

Reply:
in evaluationOutputHistory_ we store the value + constraint so its of dim 2 yes

my modification changes the dimension of the outputHistory_ of OptimResult:
```result_.store(inP, Point(1, outP[0])```